csharp/ql/src/experimental/Security Features/backdoor/PotentialTimeBomb.ql

```diff
@@ -11,32 +11,16 @@
  */
 
 import csharp
-import DataFlow
+import Flow::PathGraph
 
-query predicate nodes = PathGraph::nodes/3;
-
-query predicate edges(DataFlow::PathNode a, DataFlow::PathNode b) {
-  PathGraph::edges(a, b)
+query predicate edges(Flow::PathNode a, Flow::PathNode b) {
+  Flow::PathGraph::edges(a, b)
   or
-  exists(
-    FlowsFromGetLastWriteTimeConfigToTimeSpanArithmeticCallable conf1,
-    FlowsFromTimeSpanArithmeticToTimeComparisonCallable conf2
-  |
-    conf1 = a.getConfiguration() and
-    conf1.isSink(a.getNode()) and
-    conf2 = b.getConfiguration() and
-    b.isSource()
-  )
+  FlowsFromGetLastWriteTimeConfigToTimeSpanArithmeticCallableConfig::isSink(a.getNode()) and
+  FlowsFromTimeSpanArithmeticToTimeComparisonCallableConfig::isSource(b.getNode())
```

Comment:
This looks like it could be solved with state and a single configuration instead.

Reply:
The output of the query uses some of the intermediate sources/sinks.

Reply:
OK, let's leave it as-is for now then (it is experimental anyway).